lib/iris/tests/test_merge.py

```diff
@@ -105,6 +108,111 @@ def test_colpex(self):
+            result = cubes.merge_cube()
+            expected = self._make_data([0, 1], dtype=self.dtype)
+            self.assertArrayEqual(result.data, expected)
+            self.assertEqual(result.dtype, self.dtype)
+
+    def test__masked_masked(self):
+        for lazy0, lazy1 in self.lazy_combos:
+            cubes = iris.cube.CubeList()
+            mask = [(0,), (0,)]
+            cubes.append(self._make_cube(0, mask=mask, lazy=lazy0,
+                                         dtype=self.dtype,
+                                         fill_value=self.fill_value))
+            mask = [(1,), (1,)]
+            cubes.append(self._make_cube(1, mask=mask, lazy=lazy1,
+                                         dtype=self.dtype,
+                                         fill_value=self.fill_value))
+            result = cubes.merge_cube()
+            mask = [(0, 1), (0, 1), (0, 1)]
+            expected = self._make_data([0, 1], mask=mask, dtype=self.dtype,
+                                       fill_value=self.fill_value)
+            self.assertMaskedArrayEqual(result.data, expected)
+            self.assertEqual(result.fill_value, self.fill_value)
+            self.assertEqual(result.dtype, self.dtype)
+
+    def test__ndarray_masked(self):
+        for lazy0, lazy1 in self.lazy_combos:
+            cubes = iris.cube.CubeList()
+            cubes.append(self._make_cube(0, lazy=lazy0, dtype=self.dtype))
+            mask = [(0, 1), (0, 1)]
+            cubes.append(self._make_cube(1, mask=mask, lazy=lazy1,
+                                         dtype=self.dtype,
+                                         fill_value=self.fill_value))
+            result = cubes.merge_cube()
+            mask = [(1, 1), (0, 1), (0, 1)]
+            expected = self._make_data([0, 1], mask=mask, dtype=self.dtype,
+                                       fill_value=self.fill_value)
+            self.assertMaskedArrayEqual(result.data, expected)
+            self.assertEqual(result.fill_value, self.fill_value)
+            self.assertEqual(result.dtype, self.dtype)
+
+    def test__masked_ndarray(self):
+        for lazy0, lazy1 in self.lazy_combos:
+            cubes = iris.cube.CubeList()
+            mask = [(0, 1), (0, 1)]
+            cubes.append(self._make_cube(0, mask=mask, lazy=lazy0,
+                                         dtype=self.dtype,
+                                         fill_value=self.fill_value))
+            cubes.append(self._make_cube(1, lazy=lazy1, dtype=self.dtype))
+            result = cubes.merge_cube()
+            mask = [(0, 0), (0, 1), (0, 1)]
+            expected = self._make_data([0, 1], mask=mask, dtype=self.dtype,
+                                       fill_value=self.fill_value)
+            self.assertMaskedArrayEqual(result.data, expected)
+            self.assertEqual(result.fill_value, self.fill_value)
+            self.assertEqual(result.dtype, self.dtype)
+
+
```

Comment:
@dkillick Some neat tests for all combinations of real vs lazy and ndarray vs masked ... reasonably comprehensive now! :smile: